Dos minimal2

david4096 opened the pull request.

david4096 committed:
Add GetDataObject
It seems we lacked a basic get by id functionality. Renamed the variable name to data_object_id to be more descriptive which led to other minor changes.

david4096 committed:
Remove extraneous methods
The features underlying these methods are made possible by the data object message itself. Services that simplify the response can use underlying API methods.

david4096 committed:
Rearrange entries, change package name
The variable names weren\'t normalized to each other which was confusing. This rearranges the file to be more readable, adds message types where needed, and begins a refactor to replace the context-less \'id\' with .

david4096 committed:
Normalize id usage, add update messages.
Changes id names to be more descriptive. Adds message types for data bundle.

david4096 committed:
Remove extraneous object
Is it necessary to call it a BundleObject? Bundle is probably descriptive enough.

david4096 committed:
Remove checksum URL
All of the functionality of get-by-checksum is possible using a single endpoint. Simplifies the offering.

david4096 committed:
Rearrange bundles under objects

david4096 committed:
Add get data bundle, fix message type
Added message types for working with bundles

david4096 committed:
Improve comment style
Rearrange to place data object above data bundle.

david4096 committed:
Remove id from creation
During creation one should not specify the ID as it will be assigned by the DOS instance. To control the naming scheme aliases should be used.

david4096 committed:
Add more comments, change id search to alias
Being able to GET by ID and list using aliases seems like a more natural functionality. This makes one have to write a simple script to get all of the version of a document, but that seems like a rare edge case.

david4096 committed:
Make creation more clear
The previous creation implementation left a lot of gray area of how the messages were to be created. This adds fields to the create messages that make it clear what is required to properly form a Data Object or Bundle.

david4096 committed:
Change metadata to map of ListValue, add checksum type
Close #15 thanks @kelrott!
Support various uploading mechanisms and having multiple checksums per Data Object. Close #12

david4096 committed:
Update swagger

bwalsh commented on proto/data_objects.proto:
```diff
@@ -91,3 +85,51 @@ message DataObject {
+  // At least one checksum must be provided.
+  // The data bundle SHA256 checksum is computed over all the
+  // SHA256 checksums of the Data Objects that make up this bundle.
+  repeated Checksum checksum =  6;
```
Can this be computed by iterating over the dataobject records only?  Or, does one need to download and compute over the actual files?

david4096 replied:
For the case of Data Bundles I believe the intention was to only concatenate the checksums of the underlying data objects.

david4096 replied:
Change the default to MD5. Specify more clearly how to concatenate MD5 digests.

bwalsh reviewed (commented):
one question re. the hashes at the bundle level

david4096 committed:
Make updated optional close #16

david4096 committed:
Turn list request to a post

david4096 committed:
Basic server/client scaffolding.
Added a script to modify the swagger to support connexion and bravado properly. Need to run using sh tools/prepare_swagger.sh after running cwltool.

david4096 committed:
Change object updating semantics to be more clear
Follows the same pattern as Create

david4096 committed:
Fix URLs in readme

david4096 committed:
Add demo of client usage
Basic CRUD example in app.py

david4096 commented:
Added a basic server/client example.

david4096 committed:
Merge branch 'master' into dos-minimal2

david4096 commented on proto/data_objects.proto:
```diff
@@ -1,65 +1,54 @@
-  string version = 5;
+  // OPTIONAL
+  // The digest method used to create the checksum.If left unspecified it is
+  // assumed the SHA256 checksum was calculated over the data.
```
Make MD5 the default, change to open string.

david4096 commented on proto/data_objects.proto:
```diff
@@ -96,3 +86,51 @@ message DataObject {
+
+  // OPTIONAL
+  // A set of key-value pairs that represent system metadata about the object.
+  map<string, google.protobuf.ListValue> system_metadata = 11;
```
This might vary from bucket to bucket. Should be moved to the URL message.

david4096 commented on proto/data_objects.proto:
```diff
@@ -69,23 +58,24 @@ message DataObject {
 
   // OPTIONAL
-  // A url pointing to the object in a given container
+  // The list of URLs that can be used to access the Data Object.
```
URL to become complex object that has metadata.

david4096 committed:
Change URL to include metadata
@bwalsh points out that file metadata is unique to the blob store and URL instance.

david4096 committed:
Change checksum to string selection
@briandoconnor suggested that enum can be challenging to use and often folks just refer to the documentation. Adds the hashing type as a string and clarifies usage of the ChecksumRequest

david4096 committed:
Add basic list functionality

david4096 committed:
Index data from the current GDC
Work around a bug in how bravado handles strings. The protobuf toolchain works around JavaScript being unable to represent large numbers by wrapping them in strings. JSONSchema has different opinions about this.

david4096 committed:
Improve UpdateDataBundleRequest
Add fields to make update request more clear.

david4096 committed:
Add flake fixes and flake test
Add dev-requirements as needed for modifying the schemas, regular requirements just to run.

david4096 committed:
Add s to checksum, DEVELOP doc
Adds a development document that goes through the process of making a schema modification
Add flaketest to travis

david4096 committed:
Fix gdc alias, fix comment

david4096 committed:
Remove mention of sha256
Object stores often only report sha256.
Flake fixes

david4096 committed:
Get more gdc metadata
Allows the entirety of gdc metadata to be passed through.

david4096 committed:
Change metadata to struct, simplify create/update

david4096 committed:
Change Create/Update semantics to include Data Object
Fix errors caused by Create/Update request change
Change alias comment from identify to "find" to be more clear

david4096 committed:
Add versions endpoint
Clarifies update semantics to list all versions of a data object

david4096 committed:
Fix requirements to point at latest bravado

david4096 committed:
Demonstrate how to set ID and version
Add logic to app for allowing arbitrary ID setting if not provided
Roll back swagger link hopefully it works
Fix requirements, hopefully for real this time

david4096 committed:
Add paging functionality
A crude pager

david4096 committed:
Add setup, restructure python
Makes the package pip installable and moves the client to the ga4gh namespace.
Hack constraints so that it installs the bravado package in both setup py and pip installs.
Move controllers to their own namespace to make them easier to handle

david4096 commented on .travis.yml:
```diff
@@ -13,5 +13,8 @@ before_install:
     - bash tools/travis-install-protoc.sh 3.2.0
     - export PATH=/tmp/proto3.2.0/bin:$PATH
     - pip install -r requirements.txt
+    - pip install -r dev-requirements.txt
```
#L13

bwalsh commented on proto/data_objects_service.proto:
```diff
@@ -1,195 +1,274 @@
 }
 
-message DataBundleStatusResponse {
+message UpdateDataObjectRequest {
```
I think this might be out of sync with the underlying DataObject.
Propose that UpdateDataObject signature be:
`DataObject  updated = UpdateDataObject(DataObject update)`

david4096 committed:
Implement Data Bundles
Smooth out some schema issues, UpdateDataBundle is simpler, DataBundles include a list of DataObject IDs.

Demo updated with more example usage.

david4096 committed:
Move and spruce up examples
Remove demo from CLI entrypoint
Fix travis yml

david4096 committed:
Change update bundle semantics
Simplifies the update bundle message to accept an entire bundle.
